### PR TITLE
Revert "Revert "Move common dependencies to common-lib""

### PR DESCRIPTION
**Pull request by jonathonherbert**
Reverts guardian/typerighter#234 and includes https://github.com/guardian/typerighter/pull/235 and a bump to pan-domain-auth to include https://github.com/guardian/pan-domain-authentication/pull/114 – all of which should fix the conflicts with `google-api-client` that led to the revert.

## How to test

- [x] Run the application in CODE. It should deploy as expected.
- [x] Refresh the rules in the checker. The ingest process should work as expected.
- [x] Run a check against a document. It should produce matches as expected. 
- [x] Import rules via the management system (guidance in #229.) Run a check with these newly imported rules. 

Tested all of the above in CODE.

### CODE REVIEW

File: apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala [hunk 1]
```diff
@@ -94,19 +96,19 @@ class LanguageToolMatcherTest extends AsyncFlatSpec with Matchers {
...
   """
-  val exampleRulegroup =  LTRuleXML(
+  val exampleRulegroup =  typerighter.model.LTRuleXML(
     "EXAMPLE_RULEGROUP",
     exampleRulegroupXml,
     exampleCategory,
     "An example rulegroup with custom XML"
   )
-  val exampleBadRule1 = LTRuleXML(
+  val exampleBadRule1 = typerighter.model.LTRuleXML(
```
Comment (phillipbarron) [inline apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala:105]: Is this required?
```suggestion
  val exampleBadRule1 = LTRuleXML(
```

File: apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala [hunk 2]
```diff
@@ -94,19 +96,19 @@ class LanguageToolMatcherTest extends AsyncFlatSpec with Matchers {
...
     "EXAMPLE_RULE",
     exampleBadRuleXml,
     exampleCategory2,
     "An example rule with custom XML that contains references to nonexistent entities"
   )
-  val exampleBadRule2 = LTRuleXML(
+  val exampleBadRule2 = typerighter.model.LTRuleXML(
```
Comment (phillipbarron) [inline apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala:111]: As above
```suggestion
  val exampleBadRule2 = LTRuleXML(
```

File: apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala (hunk 1 above)
Comment (jonathonherbert) [reply apps/checker/test/scala/matchers/LanguageToolMatcherTest.scala]: Thanks for spotting! [4e54ff6](https://github.com/guardian/typerighter/pull/239/commits/4e54ff6739e8ccff015950feceb0b17ffda306f9)

**Review by phillipbarron — APPROVED** [review]
Looks good to me - I see you have tested in code 👍

**Pull request merged**